spec/system/candidate_interface/new-references/candidate_add_references_spec.rb

```diff
@@ -0,0 +1,388 @@
+    and_i_click_save_and_continue
+  end
+
+  def when_i_fill_in_my_second_references_name
+    fill_in 'What is the referee’s name?', with: 'John Doe'
+  end
+
+  def when_i_provide_a_second_valid_email_address
+    fill_in 'What is the referee’s email address?', with: 'johndoe@example.com'
+  end
+
+  def when_i_fill_in_my_second_references_relationship
+    fill_in 'How do you know this referee and how long have you known them', with: 'He is my sing teacher?'
+  end
+
+  def then_i_see_the_referee_email_page
+    expect(page).to have_current_path candidate_interface_new_references_email_address_path(@application.application_references.last.id)
+  end
+
+  def when_i_click_save_and_continue_without_providing_an_emailing
+    and_i_click_save_and_continue
+  end
+
+  def then_i_should_be_told_to_provide_an_email_address
+    expect(page).to have_content 'Enter your referee’s email address'
+  end
+
+  def and_a_validation_error_is_logged_for_blank_email_address
+    expect(ValidationError.count).to be 4
+  end
+
+  def when_i_provide_an_email_address_with_an_invalid_format
+    fill_in 'candidate-interface-reference-referee-email-address-form-email-address-field-error', with: 'invalid.email.address'
+  end
+
+  def then_i_am_told_my_email_address_needs_a_valid_format
+    expect(page).to have_content 'Enter an email address in the correct format, like name@example.com'
+  end
+
+  def and_a_validation_error_is_logged_for_invalid_email_address
+    expect(ValidationError.count).to be 5
+  end
+
+  def when_i_provide_a_valid_email_address
+    fill_in 'candidate-interface-reference-referee-email-address-form-email-address-field-error', with: 'iamtheone@whoknocks.com'
+  end
+
+  def then_i_see_the_relationship_page
+    expect(page).to have_current_path candidate_interface_new_references_relationship_path(@application.application_references.last.id)
+  end
+
+  def when_i_click_save_and_continue_without_providing_a_relationship
+    and_i_click_save_and_continue
+  end
+
+  def then_i_should_be_told_to_provide_a_relationship
+    expect(page).to have_content 'Enter how you know this referee and for how long'
+  end
+
+  def and_a_validation_error_is_logged_for_relationship
+    expect(ValidationError.count).to be 6
+  end
+
+  def when_i_fill_in_my_references_relationship
+    fill_in 'candidate-interface-reference-referee-relationship-form-relationship-field-error', with: 'Through nefarious behaviour.'
+  end
+
+  def and_i_should_see_my_references_details
+    expect(page).to have_content 'Academic'
+    expect(page).to have_content 'Walter White'
+    expect(page).to have_content 'iamtheone@whoknocks.com'
+    expect(page).to have_content 'Through nefarious behaviour.'
+  end
+
+  def when_i_click_change_on_the_references_name
+    click_link 'Change name for Walter White'
+  end
+
+  def and_i_input_a_new_name
+    fill_in 'What is the referee’s name?', with: 'Jessie Pinkman'
```

Comment:
it's spelled Jesse 🤣 

![](https://media.giphy.com/media/MBVemoHuyw9Ik/giphy.gif)